project/LookupTableGen.scala

```diff
@@ -0,0 +1,225 @@
+package org.typelevel.idna4s.build
+
+import cats._
+import cats.data._
+import cats.syntax.all._
+import scala.annotation.tailrec
+import scala.collection.immutable.SortedSet
+import scala.collection.immutable.SortedMap
+
+object LookupTableGen {
+
+  // Why is this not in cats?
+  private implicit def orderingFromOrder[A](implicit A: Order[A]): Ordering[A] =
+    A.toOrdering
+
+  def printAsHex(i: Int): String =
+    String.format("0x%04X", i.asInstanceOf[Object]) // Don't ask...
```

Comment:
Would this work? 😅 
```suggestion
    String.format("0x%04X", new Integer(i)) // Don't ask...
```

Reply:
Apparently that's deprecated, but it does look like I can use `Integer.valueOf(i))`. I'll get that cleaned up.